diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/BlockStructureBraces.kt

```diff
@@ -77,9 +77,11 @@ class BlockStructureBraces : Rule("block-structure") {
     }
 
     private fun checkClass(node: ASTNode, configuration: BlockStructureBracesConfiguration) {
-        if (node.hasChildOfType(CLASS_BODY)) {
-            checkOpenBraceOnSameLine(node, CLASS_BODY, configuration)
-            checkCloseBrace(node.findChildByType(CLASS_BODY)!!, configuration)
+        if (!isNodeEmpty(node.findChildByType(CLASS_BODY))) {
+            if (node.hasChildOfType(CLASS_BODY)) {
```

Comment:
```suggestion
        if (node.hasChildOfType(CLASS_BODY) && !isNodeEmpty(node.findChildByType(CLASS_BODY))) {
```
No need to make nested conditions here. Also, seems strange that firstly you do something with `CLASS_BODY` node and only then check if it is present. If your `isNodeEmpty` method covers case when argument is null, you might as well remove `hasChildOfType` (actually, you do exactly this in `checkFun`).